docs/airflow/README.md

```diff
@@ -1,5 +1,209 @@
+]
+```
+
+- The 'extract' logic for each data source is specified in `airflow/include/tasks/extract.py`.  These functions can be copied/pasted to select new sources.
+For instance, the `extract_astro_blogs` function shows an example of using the Python [requests](https://pypi.org/project/requests/) and [Beautiful Soup](https://pypi.org/project/beautifulsoup4/) libraries to extract blog posts from http://astronomer.io/blog.  This code can be adapted to ingeset from other sources.
+
+11. Ingest to Test: A free Weaviate Sandbox cluster can be created with a trial account. The sandbox is a free 14-day temporary cluster. After creating the sandbox cluster note the `Cluster URL` and the Admin API Key.
+- Open the `airflow/.env` file and update the following:
+-- `ASK_ASTRO_ENV`: Set it to `test`
+-- `AIRFLOW_CONN_WEAVIATE_TEST`: Update the host and token with the sandbox settings. The OpenAI key should be the same as in the 'local' setting.
+- Restart astro with the new environment variables.
+```bash
+astro dev restart
+```
+- Rerun the bulk ingest DAG
+```bash
+astro dev run dags unpause ask_astro_load_bulk
+astro dev run dags trigger ask_astro_load_bulk
+```
+
+This time when the DAG runs it will use the serialized extracted documents which will run much faster. Only the `import_data` tasks will take time. Look in the task logs for this task to note that it is now using the 'test' cluster. For example:
+```
+[2023-10-17, 12:26:21 UTC] {base.py:73} INFO - Using connection ID 'weaviate_test' for task execution.
+```
+
+As the DAG runs you can see the document count increasing in the [Weaviate Console](https://console.weaviate.cloud) with a query such as:
+```
+{ Aggregate { Docs { meta { count } } } }
+```
+
+After integration testing with the front end the sandbox cluster can be removed.
+
+12. Promote to production: Step 10 showed an example of switching between different Weaviate instances. Follow the same approach to point Airflow at a 'production' (ie. non-temporary sandbox). Perform the following to promote from Airflow running locally to a production deployment in Astro.
+- Create a free [Astro trial](https://docs.astronomer.io/astro/trial) account, organization and deployment.
+- [Deploy](https://docs.astronomer.io/astro/deploy-code) the Astro project
+```bash
+astro login
+```
+```bash
+astro deployment create -n 'ask astro dev'
+astro deployment variable update -lsn 'ask astro dev'
+astro deployment variable update -n 'ask astro dev' ASK_ASTRO_ENV=dev
+astro deploy -fn 'ask astro dev'
+```
+
+13. Query the Data: A simple Streamlit application is provided to query the data. For this application you will have needed to provide an OpenAI key with the Weaviate connection string in step 3.
+
+- Connect to the webserver container with the Astro CLI
+```bash
+astro dev bash -w
+```
+
+- Start Streamlit
+```bash
+cd include/streamlit
+python -m streamlit run ./streamlit_app.py
+```
 
-- [astro-cli](https://docs.astronomer.io/astro/cli/overview)
```

Comment:
This is required as a pre-requisite to run this. It should be there

Reply:
Thanks! Just added it back.